File: src/main/java/me/glaremasters/guilds/configuration/sections/GuildBuffSettings.java
/*
 * MIT License
 *
 * Copyright (c) 2018 Glare
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package me.glaremasters.guilds.configuration.sections;

import ch.jalu.configme.Comment;
import ch.jalu.configme.SettingsHolder;
import ch.jalu.configme.properties.Property;

import java.util.List;

import static ch.jalu.configme.properties.PropertyInitializer.newListProperty;
import static ch.jalu.configme.properties.PropertyInitializer.newProperty;

/**
 * Created by GlareMasters
 * Date: 1/17/2019
 * Time: 2:29 PM
 */
public class GuildBuffSettings implements SettingsHolder {

    private static final String BUFF_PATH = "guis.guild-buffs.buffs.";

    @Comment("What should the name of the inventory be?")
    public static final Property<String> GUILD_BUFF_NAME =
            newProperty("guis.guild-buffs.gui-name", "Guild Buffs");

    @Comment("Do we want to allow users to have more than one buff at a time?")
    public static final Property<Boolean> BUFF_STACKING =
            newProperty("guis.guild-buffs.buff-stacking", false);

    @Comment("What do you want to name the buff?")
    public static final Property<String> HASTE_NAME =
            newProperty(BUFF_PATH + "haste.name", "&a&lSubstance of the Redmod Graff");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> HASTE_PRICE =
            newProperty(BUFF_PATH + "haste.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> HASTE_TIME =
            newProperty(BUFF_PATH + "haste.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> HASTE_AMPLIFIER =
            newProperty(BUFF_PATH + "haste.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> HASTE_ICON =
            newProperty(BUFF_PATH + "haste.icon", "FEATHER");

    @Comment("What type of potion is this?")
    public static final Property<String> HASTE_TYPE =
            newProperty(BUFF_PATH + "haste.type", "FAST_DIGGING");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> HASTE_LORE =
            newListProperty(BUFF_PATH + "haste.description", "&aType » &7Haste", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> HASTE_DISPLAY =
            newProperty(BUFF_PATH + "haste.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> SPEED_NAME =
            newProperty(BUFF_PATH + "speed.name", "&a&lBlessing of the Cheetah");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> SPEED_PRICE =
            newProperty(BUFF_PATH + "speed.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> SPEED_TIME =
            newProperty(BUFF_PATH + "speed.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> SPEED_AMPLIFIER =
            newProperty(BUFF_PATH + "speed.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> SPEED_ICON =
            newProperty(BUFF_PATH + "speed.icon", "SUGAR");

    @Comment("What type of potion is this?")
    public static final Property<String> SPEED_TYPE =
            newProperty(BUFF_PATH + "speed.type", "SPEED");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> SPEED_LORE =
            newListProperty(BUFF_PATH + "speed.description", "&aType » &7Speed", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> SPEED_DISPLAY =
            newProperty(BUFF_PATH + "speed.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> FR_NAME =
            newProperty(BUFF_PATH + "fire-resistance.name", "&a&lScales of the Dragon");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> FR_PRICE =
            newProperty(BUFF_PATH + "fire-resistance.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> FR_TIME =
            newProperty(BUFF_PATH + "fire-resistance.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> FR_AMPLIFIER =
            newProperty(BUFF_PATH + "fire-resistance.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> FR_ICON =
            newProperty(BUFF_PATH + "fire-resistance.icon", "BLAZE_POWDER");

    @Comment("What type of potion is this?")
    public static final Property<String> FR_TYPE =
            newProperty(BUFF_PATH + "fire-resistance.type", "FIRE_RESISTANCE");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> FR_LORE =
            newListProperty(BUFF_PATH + "fire-resistance.description", "&aType » &7Fire-Resistance", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> FR_DISPLAY =
            newProperty(BUFF_PATH + "fire-resistance.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> NV_NAME =
            newProperty(BUFF_PATH + "night-vision.name", "&a&lEyes of the Lurking Demon");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> NV_PRICE =
            newProperty(BUFF_PATH + "night-vision.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> NV_TIME =
            newProperty(BUFF_PATH + "night-vision.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> NV_AMPLIFIER =
            newProperty(BUFF_PATH + "night-vision.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> NV_ICON =
            newProperty(BUFF_PATH + "night-vision.icon", "REDSTONE_TORCH_ON");

    @Comment("What type of potion is this?")
    public static final Property<String> NV_TYPE =
            newProperty(BUFF_PATH + "night-vision.type", "NIGHT_VISION");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> NV_LORE =
            newListProperty(BUFF_PATH + "night-vision.description", "&aType » &7Night-Vision", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> NV_DISPLAY =
            newProperty(BUFF_PATH + "night-vision.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> INVISIBILITY_NAME =
            newProperty(BUFF_PATH + "invisibility.name", "&a&lFeet of the Ghostly Walker");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> INVISIBILITY_PRICE =
            newProperty(BUFF_PATH + "invisibility.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> INVISIBILITY_TIME =
            newProperty(BUFF_PATH + "invisibility.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> INVISIBILITY_AMPLIFIER =
            newProperty(BUFF_PATH + "invisibility.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> INVISIBILITY_ICON =
            newProperty(BUFF_PATH + "invisibility.icon", "EYE_OF_ENDER");

    @Comment("What type of potion is this?")
    public static final Property<String> INVISIBILITY_TYPE =
            newProperty(BUFF_PATH + "invisibility.type", "INVISIBILITY");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> INVISIBILITY_LORE =
            newListProperty(BUFF_PATH + "invisibility.description", "&aType » &7Invisibility", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> INVISIBILITY_DISPLAY =
            newProperty(BUFF_PATH + "invisibility.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> STRENGTH_NAME =
            newProperty(BUFF_PATH + "strength.name", "&a&lMighty Strength of the Pouncing Lion");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> STRENGTH_PRICE =
            newProperty(BUFF_PATH + "strength.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> STRENGTH_TIME =
            newProperty(BUFF_PATH + "strength.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> STRENGTH_AMPLIFIER =
            newProperty(BUFF_PATH + "strength.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> STRENGTH_ICON =
            newProperty(BUFF_PATH + "strength.icon", "DIAMOND_SWORD");

    @Comment("What type of potion is this?")
    public static final Property<String> STRENGTH_TYPE =
            newProperty(BUFF_PATH + "strength.type", "INCREASE_DAMAGE");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> STRENGTH_LORE =
            newListProperty(BUFF_PATH + "strength.description", "&aType » &7Strength", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> STRENGTH_DISPLAY =
            newProperty(BUFF_PATH + "strength.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> JUMP_NAME =
            newProperty(BUFF_PATH + "jump.name", "&a&lBounce of the Quick Witted Rabbit");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> JUMP_PRICE =
            newProperty(BUFF_PATH + "jump.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> JUMP_TIME =
            newProperty(BUFF_PATH + "jump.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> JUMP_AMPLIFIER =
            newProperty(BUFF_PATH + "jump.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> JUMP_ICON =
            newProperty(BUFF_PATH + "jump.icon", "DIAMOND_BOOTS");

    @Comment("What type of potion is this?")
    public static final Property<String> JUMP_TYPE =
            newProperty(BUFF_PATH + "jump.type", "JUMP_BOOST");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> JUMP_LORE =
            newListProperty(BUFF_PATH + "jump.description",  "&aType » &7Jump", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> JUMP_DISPLAY =
            newProperty(BUFF_PATH + "jump.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> WB_NAME =
            newProperty(BUFF_PATH + "water-breathing.name", "&a&lLungs of the Albino Shark");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> WB_PRICE =
            newProperty(BUFF_PATH + "water-breathing.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> WB_TIME =
            newProperty(BUFF_PATH + "water-breathing.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> WB_AMPLIFIER =
            newProperty(BUFF_PATH + "water-breathing.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> WB_ICON =
            newProperty(BUFF_PATH + "water-breathing.icon", "BUCKET");

    @Comment("What type of potion is this?")
    public static final Property<String> WB_TYPE =
            newProperty(BUFF_PATH + "water-breathing.type", "WATER_BREATHING");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> WB_LORE =
            newListProperty(BUFF_PATH + "water-breathing.description", "&aType » &7Water-Breathing", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> WB_DISPLAY =
            newProperty(BUFF_PATH + "water-breathing.display", true);

    @Comment("What do you want to name the buff?")
    public static final Property<String> REGENERATION_NAME =
            newProperty(BUFF_PATH + "regeneration.name", "&a&lIntegrity of the Mystic Witch");

    @Comment("How much do you want the buff to cost?")
    public static final Property<Double> REGENERATION_PRICE =
            newProperty(BUFF_PATH + "regeneration.price", 60.0);

    @Comment("How long (in second) should the buff last?")
    public static final Property<Integer> REGENERATION_TIME =
            newProperty(BUFF_PATH + "regeneration.time", 60);

    @Comment("How strong do you want the buff to be? 0 = Potion Level 1, 1 = Potion Level 2, etc...")
    public static final Property<Integer> REGENERATION_AMPLIFIER =
            newProperty(BUFF_PATH + "regeneration.amplifier", 0);

    @Comment("What item do you want to represent the buff?")
    public static final Property<String> REGENERATION_ICON =
            newProperty(BUFF_PATH + "regeneration.icon", "EMERALD");

    @Comment("What type of potion is this?")
    public static final Property<String> REGENERATION_TYPE =
            newProperty(BUFF_PATH + "regeneration.type", "REGENERATION");

    @Comment("You can put as much as you want here")
    public static final Property<List<String>> REGENERATION_LORE =
            newListProperty(BUFF_PATH + "regeneration.description", "&aType » &7Regeneration", "&aLength » &760 Seconds", "&aCost » &7$60");

    @Comment("Do you want this buff to show in-game?")
    public static final Property<Boolean> REGENERATION_DISPLAY =
            newProperty(BUFF_PATH + "regeneration.display", true);

    @Comment("What do you want the name of the Vault to be?")
    public static final Property<String> VAULT_NAME =
            newProperty("guis.vault.name", "Guild Vault");

    private GuildBuffSettings() {
    }

}

File: src/main/kotlin/me/glaremasters/guilds/conf/GuildBuffSettings.kt
/*
 * MIT License
 *
 * Copyright (c) 2023 Glare
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package me.glaremasters.guilds.conf

import ch.jalu.configme.Comment
import ch.jalu.configme.SettingsHolder
import ch.jalu.configme.properties.Property
import ch.jalu.configme.properties.PropertyInitializer
import ch.jalu.configme.properties.PropertyInitializer.newBeanProperty
import ch.jalu.configme.properties.PropertyInitializer.newProperty
import ch.jalu.configme.properties.types.BeanPropertyType
import me.glaremasters.guilds.conf.objects.BuffCommand
import me.glaremasters.guilds.conf.objects.BuffNav
import me.glaremasters.guilds.conf.objects.BuffNavItem
import me.glaremasters.guilds.conf.objects.BuffSettings
import me.glaremasters.guilds.conf.objects.GuildBuff

internal object GuildBuffSettings : SettingsHolder {

    @JvmField
    @Comment("What should the name of the inventory be?")
    val GUI_NAME: Property<String> = newProperty("guild-buffs.gui-name", "Guild Buffs")

    @JvmField
    @Comment("How often (in seconds) can a guild buy a buff?")
    val COOLDOWN = newProperty("guild-buffs.cooldown", 60)

    @JvmField
    @Comment("Do we want to allow users to have more than one buff at a time?")
    val BUFF_STACKING: Property<Boolean> = newProperty("guild-buffs.buff-stacking", false)

    @JvmField
    @Comment("Set the name and material for the navigation buttons")
    val NAVIGATION: Property<BuffNav> = newBeanProperty(BuffNav::class.java, "guild-buffs.nav", BuffNav(BuffNavItem("EMPTY_MAP", "Next"), BuffNavItem("EMPTY_MAP", "Previous")))

    @JvmField
    @Comment("This is where the buffs themselves are to be created. You can create an unlimited number of buffs as long as the IDENTIFIERS are different.",
            "Ths identifiers can be anything you want, they just tell the plugin that they aren't the same as the buff before.",
            "The buffs are loaded into the menu in the order they are listed in the config. So, to change the order, simply change their location in their list of the config.",
            "Example Buff:",
            "    # The identifier of the buff, can be anything you want.\n" +
                    "#    - identifier: '1'\n" +
                    "    # This is what the buff will look like to the player when they DON'T have the permission listed below.\n" +
                    "#      locked:\n" +
                    "#        name: '&a&lSubstance of the Redmod Graff'\n" +
                    "#        material: FEATHER\n" +
                    "#        lore:\n" +
                    "#        - '&aType » &7Haste'\n" +
                    "#        - '&aLength » &760 Seconds'\n" +
                    "#        - '&aCost » &7\$60'\n" +
                    "    # This is what the buff will look like to the player when they DO have the permission listed below.\n" +
                    "#      unlocked:\n" +
                    "#        name: '&a&lSubstance of the Redmod Graff'\n" +
                    "#        material: FEATHER\n" +
                    "#        lore:\n" +
                    "#        - '&aType » &7Haste'\n" +
                    "#        - '&aLength » &760 Seconds'\n" +
                    "#        - '&aCost » &7\$60'\n" +
                    "    # The price of the buff, pulled from Guild Bank.\n" +
                    "#      price: 60.0\n" +
                    "    # The effects that the guild will get for buying the buff. You should be able to list as many as you want.\n" +
                    "    # The way you do it is: EFFECT_TYPE;AMPLIFICATION;LENGTH OF BUFF\n" +
                    "    # So, this effect here will give Haste I for 60 seconds.\n" +
                    "#      effects:\n" +
                    "#      - FAST_DIGGING;0;60\n" +
                    "    # The permission required to purchase the buff.\n" +
                    "#      permission: example.perm.here\n" +
                    "    # Would you like to execute commands on the player that bought the buff for the guild? Supports {player}, {buyer}, {buff_name}.\n" +
                    "#      clicker:\n" +
                    "#        enabled: false\n" +
                    "#        commands:\n" +
                    "#        - ''\n" +
                    "    # Would you like to execute commands on the guild that bough the buff? Supports {player}, {buyer}, {buff_name}.\n" +
                    "#      guild:\n" +
                    "#        enabled: false\n" +
                    "#        commands:\n" +
                    "#        - ''"
    )
    val BUFFS: Property<MutableList<GuildBuff>> = PropertyInitializer.listProperty(BeanPropertyType.of(GuildBuff::class.java)).path("guild-buffs.buffs").defaultValue(
            listOf(
                    GuildBuff("1", BuffSettings("&a&lSubstance of the Redmod Graff", "FEATHER", listOf("&aType » &7Haste", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lSubstance of the Redmod Graff", "FEATHER", listOf("&aType » &7Haste", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("FAST_DIGGING;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("2", BuffSettings("&a&lBlessing of the Cheetah", "SUGAR", listOf("&aType » &7Speed", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lBlessing of the Cheetah", "SUGAR", listOf("&aType » &7Speed", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("SPEED;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("3", BuffSettings("&a&lScales of the Dragon", "BLAZE_POWDER", listOf("&aType » &7Fire-Resistance", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lScales of the Dragon", "BLAZE_POWDER", listOf("&aType » &7Fire-Resistance", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("FIRE_RESISTANCE;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("4", BuffSettings("&a&lEyes of the Lurking Demon", "EYE_OF_ENDER", listOf("&aType » &7Night-Vision", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lEyes of the Lurking Demon", "EYE_OF_ENDER", listOf("&aType » &7Night-Vision", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("NIGHT_VISION;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("5", BuffSettings("&a&lFeet of the Ghostly Walker", "DIAMOND_BOOTS", listOf("&aType » &7Invisibility", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lFeet of the Ghostly Walker", "DIAMOND_BOOTS", listOf("&aType » &7Invisibility", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("INVISIBILITY;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("6", BuffSettings("&a&lMighty Strength of the Pouncing Lion", "DIAMOND_SWORD", listOf("&aType » &7Strength", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lMighty Strength of the Pouncing Lion", "DIAMOND_SWORD", listOf("&aType » &7Strength", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("INCREASE_DAMAGE;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("7", BuffSettings("&a&lBounce of the Quick Witted Rabbit", "DIAMOND_BOOTS", listOf("&aType » &7Jump", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lBounce of the Quick Witted Rabbit", "DIAMOND_BOOTS", listOf("&aType » &7Jump", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("JUMP;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("8", BuffSettings("&a&lLungs of the Albino Shark", "BUCKET", listOf("&aType » &7Water-Breathing", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lLungs of the Albino Shark", "BUCKET", listOf("&aType » &7Water-Breathing", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("WATER_BREATHING;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf(""))),
                    GuildBuff("9", BuffSettings("&a&lIntegrity of the Mystic Witch", "EMERALD", listOf("&aType » &7Regeneration", "&aLength » &760 Seconds", "&aCost » &7$60")), BuffSettings("&a&lIntegrity of the Mystic Witch", "EMERALD", listOf("&aType » &7Regeneration", "&aLength » &760 Seconds", "&aCost » &7$60")), 60.00, listOf("REGENERATION;0;60"), "example.perm.here", BuffCommand(false, listOf("")), BuffCommand(false, listOf("")))
            )
    ).build()
}
